crates/core/tedge_api/src/lib.rs

```diff
@@ -744,195 +855,145 @@ mod tests {
+                                "action": "install"
+                            },
+                            {
+                                "name": "collectd",
+                                "version": "5.7",
+                                "url": "https://collectd.org/download/collectd-tarballs/collectd-5.12.0.tar.bz2",
+                                "action": "install"
+                            }
+                        ]
+                    },
+                    {
+                        "type": "docker",
+                        "modules": [
+                            {
+                                "name": "nginx",
+                                "version": "1.21.0",
+                                "action": "install"
+                            },
+                            {
+                                "name": "mongodb",
+                                "version": "4.4.6",
+                                "action": "remove"
+                            }
+                        ]
+                    }
+                ],
+                "failures":[
+                    {
+                        "type":"debian",
+                        "modules": [
+                            {
+                                "name":"collectd",
+                                "version":"5.7",
+                                "action":"install",
+                                "reason":"Network timeout"
+                            }
+                        ]
+                    },
+                    {
+                        "type":"docker",
+                        "modules": [
+                            {
+                                "name": "mongodb",
+                                "version": "4.4.6",
+                                "action":"remove",
+                                "reason":"Other components dependent on it"
+                            }
+                        ]
+                    }
+                ]
+            }"#;
+
+        let actual_json = response.payload.to_json();
         assert_eq!(
             remove_whitespace(&actual_json),
             remove_whitespace(expected_json)
         );
     }
-
-    #[test]
-    fn using_a_software_update_response() {
-        let json_response = r#"{
-            "id": "123",
-            "status":"failed",
-            "reason":"2 errors: fail to install [ collectd ] fail to remove [ mongodb ]",
-            "currentSoftwareList": [
-                {
-                    "type": "debian",
-                    "modules": [
-                        {
-                            "name": "nodered",
-                            "version": "1.0.0"
-                        }
-                    ]
-                },
-                {
-                    "type": "docker",
-                    "modules": [
+    /*
+
+        #[test]
+        fn using_a_software_update_response() {
```

Comment:
Remove that test? Looks like an irrelevant test as software update response doesn't carry the software list anymore.